kasa/iot/iotstrip.py

```diff
@@ -148,7 +148,7 @@ def on_since(self) -> datetime | None:
         if self.is_off:
             return None
 
-        return max(plug.on_since for plug in self.children if plug.on_since is not None)
+        return min(plug.on_since for plug in self.children if plug.on_since is not None)
```

Comment:
This changes the behavior.

Reply:
It does although this is the right behaviour :smile:.